File: balanced-binary-tree.cpp
//Runtime: 9 ms
class Solution {
public:

    int ans = 1;
    int traverse(TreeNode* root, int h)
    {
        if(!root)
            return h;

        int a = traverse(root->left, h+1);
        int b = traverse(root->right, h+1);

        if(abs(a-b) > 1)
            ans = 0;

        return max(a,b);
    }

    bool isBalanced(TreeNode* root) {
        traverse(root,0);
        return ans;
    }
};

File: balanced-binary-tree.py
# Runtime: 77 ms
class Solution(object):
    def solve(self, root):
        if root == None:
            return 0

        l = self.solve(root.left)
        r = self.solve(root.right)

        if l == -1 or r == -1 or abs(l - r) > 1:
            return -1

        return 1 + max(l, r)

    def isBalanced(self, root):
        return self.solve(root) != -1
